Setup CI

piggyback on the configuration here by @cclauss https://github.com/internetarchive/wcdimportbot/blob/main/.github/workflows/lint_python.yml

% `flake8 . --count --select=E9,F63,F7,F82,Y --show-source --statistics`
```
./asseeibot/asseeibot/models/ontology_dataframe.py:25:24: F821 undefined name 'Dataset'
            dataframe: Dataset["item", "itemLabel", "alias"] = pd.read_pickle("ontology.pkl")
                       ^
./asseeibot/asseeibot/models/ontology_dataframe.py:25:32: F821 undefined name 'item'
            dataframe: Dataset["item", "itemLabel", "alias"] = pd.read_pickle("ontology.pkl")
                               ^
./asseeibot/asseeibot/models/ontology_dataframe.py:25:40: F821 undefined name 'itemLabel'
            dataframe: Dataset["item", "itemLabel", "alias"] = pd.read_pickle("ontology.pkl")
                                       ^
./asseeibot/asseeibot/models/ontology_dataframe.py:25:53: F821 undefined name 'alias'
            dataframe: Dataset["item", "itemLabel", "alias"] = pd.read_pickle("ontology.pkl")
                                                    ^
4     F821 undefined name 'Dataset'
4
```

--- a/.github/workflows/lint_python.yml
+++ b/.github/workflows/lint_python.yml
@@ -1,0 +1,25 @@
+name: lint_python
+on: [pull_request, push]
+jobs:
+  lint_python:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+      - run: pip install --upgrade pip wheel
+      - run: pip install bandit black codespell flake8 flake8-2020 flake8-bugbear
+                         flake8-comprehensions isort mypy pytest pyupgrade safety
+      - run: bandit --recursive --skip B101 .  # B101 is assert statements
+      - run: black --check . || true
+      - run: codespell  # --ignore-words-list="" --skip="*.css,*.js,*.lock"
+      - run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+      - run: flake8 . --count --exit-zero --max-complexity=10 --max-line-length=88
+                      --show-source --statistics
+      - run: isort --check-only --profile black . || true
+      - run: pip install -r requirements.txt || pip install --editable .  || pip install . || true
+      - run: mkdir --parents --verbose .mypy_cache
+      - run: mypy --ignore-missing-imports --install-types --non-interactive . || true
+      - run: pytest . || true
+      - run: pytest --doctest-modules . || true
+      - run: shopt -s globstar && pyupgrade --py36-plus **/*.py || true
+      - run: safety check

--- a/asseeibot/models/ontology_dataframe.py
+++ b/asseeibot/models/ontology_dataframe.py
@@ -24,7 +24,7 @@ class Dataframe:
             # noinspection PyUnresolvedReferences
             try:
                 dataframe: Dataset["item", "itemLabel", "alias"] = pd.read_pickle("ontology.pkl")
-            except:
+            except NameError:
                 raise
             # This is needed for the fuzzymatching to work properly
             asseeibot.runtime_variables.ontology_dataframe = dataframe.fillna('')

--- a/asseeibot/models/ontology_dataframe.py
+++ b/asseeibot/models/ontology_dataframe.py
@@ -22,7 +22,10 @@ class Dataframe:
         if asseeibot.runtime_variables.ontology_dataframe is None:
             # This pickle is ~4MB in size and takes less than a second to load.
             # noinspection PyUnresolvedReferences
-            dataframe: Dataset["item", "itemLabel", "alias"] = pd.read_pickle("ontology.pkl")
+            try:
+                dataframe: Dataset["item", "itemLabel", "alias"] = pd.read_pickle("ontology.pkl")
+            except:
+                raise
             # This is needed for the fuzzymatching to work properly
             asseeibot.runtime_variables.ontology_dataframe = dataframe.fillna('')
 

--- a/config.example.py
+++ b/config.example.py
@@ -4,7 +4,7 @@ import logging
 # This file should be copied to config.py in the same directory
 
 bot_username = ""
-password = ""
+password = ""  # nosec: B105
 
 # User settings
 crossref_polite_pool_email = None

